lib/usb/usb_standard.c

```diff
@@ -174,6 +174,20 @@ static int usb_standard_get_descriptor(usbd_device *usbd_dev,
 				      sizeof(sd->wData[0]);
 
 			*len = MIN(*len, sd->bLength);
+		} else if (descr_idx == usbd_dev->extra_string_idx) {
+			/* This string is returned as UTF16, hence the
+			 * multiplication
+			 */
+			sd->bLength = strlen(usbd_dev->extra_string) * 2 +
```

Comment:
If you're going to let people register custom strings, and those are proper usb strings, you probably don't want to rely on strlen either.